canonical/vanilla-framework
# Add color, font, and opacity Design Tokens from Core Components Figma file

## Done

- Added new color, font, and opacity Design Tokens from [Core Components Figma file](https://www.figma.com/design/kwRMtSCI8m3lQipDeusEoI/%F0%9F%92%A0-Vanilla---Core-component-library-(Copy)?node-id=1-8&t=qShynCPuMuYCEmg8-1)

Note that any build tasks related to token compilation or updating Vanilla to use tokens will occur in separate PRs. This PR is intended to be submitted first to validate our approach can be used in Figma - I have already confirmed these tokens can be properly compiled down to usable Sass variables for use in Vanilla.

Fixes [WD-14858](https://warthogs.atlassian.net/browse/WD-14858)

## QA

- Open [Core Components Figma file](https://www.figma.com/design/kwRMtSCI8m3lQipDeusEoI/%F0%9F%92%A0-Vanilla---Core-component-library-(Copy)?node-id=1-8&t=qShynCPuMuYCEmg8-1)
- Compare local variables in color, font, and opacity categories in Figma file to values in token JSON files
- Ensure there is a matching token for each local variable in Figma file (caveat: `dimensions` and `baseline spacing` categories are not represented here and will be included in a separate PR)

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)


[WD-14858]: https://warthogs.atlassian.net/browse/WD-14858?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
